src/dlr_tvm.cc

```diff
@@ -57,27 +69,68 @@ void TVMModel::SetupTVMModule(std::vector<std::string> model_path) {
                     "to override TVM allocations. Using default allocators.";
   }
 
-  ModelPath paths = GetTvmPaths(model_path);
-  std::ifstream jstream(paths.model_json);
-  std::stringstream json_blob;
-  json_blob << jstream.rdbuf();
-  std::ifstream pstream(paths.params, std::ios::in | std::ios::binary);
-  DLRStringStream param_blob;
-  param_blob << pstream.rdbuf();
-  auto param_data = param_blob.str();
-
+  std::string graph_str;
+  std::basic_string<char, std::char_traits<char>, dlr::DLRAllocator<char>> params_str;
+  const char* params_data = nullptr;
+  size_t params_size = 0;
+  std::string model_lib_path;
+  std::string metadata_path;
+  for (DLRModelElem el : model_elems) {
+    if (el.type == DLRModelElemType::TVM_GRAPH) {
+      if (el.path != nullptr) {
+        std::ifstream graph_stream(el.path);
+        std::stringstream graph_blob;
+        graph_blob << graph_stream.rdbuf();
+        graph_str = graph_blob.str();
+      } else if (el.data != nullptr) {
+        const char* graph_cstr = static_cast<const char*>(el.data);
+        graph_str = graph_cstr;
+      } else {
+        throw dmlc::Error("Invalid TVM model element TVM_GRAPH");
+      }
+    } else if (el.type == DLRModelElemType::TVM_PARAMS) {
+      if (el.path != nullptr) {
+        std::ifstream pstream(el.path, std::ios::in | std::ios::binary);
+        DLRStringStream params_blob;
+        params_blob << pstream.rdbuf();
+        params_str = params_blob.str();
+        params_data = params_str.data();
+        params_size = params_str.size();
+      } else if (el.data != nullptr && el.data_size > 0) {
+        params_data = static_cast<const char*>(el.data);
+        params_size = el.data_size;
+      } else {
+        throw dmlc::Error("Invalid TVM model element TVM_PARAMS");
+      }
+    } else if (el.type == DLRModelElemType::TVM_LIB) {
+      if (el.path != nullptr) {
+        model_lib_path = el.path;
+      } else {
+        throw dmlc::Error("Invalid TVM model element TVM_LIB");
+      }
+    } else if (el.type == DLRModelElemType::NEO_METADATA) {
+      if (el.path != nullptr) {
+        metadata_path = el.path;
+      } else {
```

Comment:
same comment here, metadata could be set as data or a path

Reply:
fixed